identity-service/src/typed-routes/solana/solanaRelayChecks.ts

```diff
@@ -0,0 +1,349 @@
+      )
+    }
+  } else {
+    throw new InvalidRelayInstructionError(
+      instructionIndex,
+      'Unsupported Associated Token Program instruction'
+    )
+  }
+}
+
+/**
+ * Allow normal transfers provided they are into userbanks, and allow
+ * close account instructions.
+ */
+const assertAllowedTokenProgramInstruction = async (
+  instructionIndex: number,
+  instruction: TransactionInstruction,
+  wallet?: string
+) => {
+  const decodedInstruction = decodeInstruction(instruction)
+  if (isTransferCheckedInstruction(decodedInstruction)) {
+    if (!wallet) {
+      throw new InvalidRelayInstructionError(
+        instructionIndex,
+        `Transfer Checked requires authentication`
+      )
+    }
+    const destination = decodedInstruction.keys.destination.pubkey
+    const userbank = await (
+      await audiusLibsWrapper.getAudiusLibsAsync()
+    ).solanaWeb3Manager!.deriveUserBank({ ethAddress: wallet, mint: 'usdc' })
+    if (!destination.equals(userbank)) {
+      throw new InvalidRelayInstructionError(
+        instructionIndex,
+        `Invalid destination account: ${destination.toBase58()}`
+      )
+    }
+  } else if (!isCloseAccountInstruction(decodedInstruction)) {
+    throw new InvalidRelayInstructionError(
+      instructionIndex,
+      'Unsupported Token Program instruction'
+    )
+  }
+}
+
+/**
+ * Checks that the Reward Manager account on the Reward Manager program
+ * matches the one in use on our deployed program.
+ */
+const assertAllowedRewardsManagerProgramInstruction = (
+  instructionIndex: number,
+  instruction: TransactionInstruction
+) => {
+  const decodedInstruction = decodeRewardManagerInstruction(instruction)
+  const rewardManager = decodedInstruction.keys.rewardManager.pubkey.toBase58()
+  if (rewardManager !== REWARD_MANAGER) {
+    throw new InvalidRelayInstructionError(
+      instructionIndex,
+      `Invalid Reward Manager Account: ${rewardManager}`
+    )
+  }
+}
+
+/**
+ * Checks that the claimable token authority matches the one in use on our
+ * deployed program. Optionally also check the user's social proof for transfers.
+ */
+const assertAllowedClaimableTokenProgramInstruction = async (
+  instructionIndex: number,
+  instruction: TransactionInstruction,
+  user?: { blockchainUserId?: number; handle?: string },
+  socialProofEnabled = false
+) => {
+  const decodedInstruction = decodeClaimableTokenInstruction(instruction)
+  const authority = decodedInstruction.keys.authority.pubkey
+  if (
+    !authority.equals(claimableTokenAuthorities[usdcMintAddress]) &&
+    !authority.equals(claimableTokenAuthorities[audioMintAddress])
+  ) {
+    throw new InvalidRelayInstructionError(
+      instructionIndex,
+      `Invalid Claimable Token Authority: ${authority}`
+    )
+  }
+
+  // Ensure user has social proof before withdrawing
+  // Currently disabled and removed from the relay flow
+  // TODO: Fix this so that tipping is unaffected
```

Comment:
Do any of these TODOs need to be TODONE before merging? Or should we at least attach tickets to these?

Reply:
https://linear.app/audius/issue/PAY-1941/clean-up-or-re-add-social-proof